File: src/com/juhnowski/onthology/AtomicFact.java
package com.juhnowski.onthology;

public class AtomicFact extends Fact{
    public  boolean isExist;
    public StatesOfAffairs statesOfAffairs;
    public Structure<Configuration<TheObject>> structure;

    public AtomicFact(){
        isTheCase = true;
    }

    public AtomicFact(Structure structure){
        super();
        this.structure = structure;
    }

}

File: src/com/juhnowski/onthology/AtomicFact.kt
package com.juhnowski.onthology

abstract class AtomicFact : Fact {
    var isExist: Boolean = false
    var statesOfAffairs: StatesOfAffairs? = null
    var structure: Structure<Configuration<TheObject>>

    abstract val picture: Thought<*>

    constructor() {
        isTheCase = true
    }

    constructor(structure: Structure<*>) : super() {
        this.structure = structure
    }

}
